src/components/vm/disks/diskAdd.jsx

```diff
@@ -365,511 +358,507 @@ const CustomPath = ({ idPrefix, onValueChanged, device, validationFailed, hideDe
-            updateDisks: false,
-            serial: "",
-        };
-    }
+const sortFunction = (poolA, poolB) => poolA.name.localeCompare(poolB.name);
+
+export const AddDiskModalBody = ({ disk, idPrefix, isMediaInsertion, vm, vms, supportedDiskBusTypes }) => {
+    const [customDiskVerificationFailed, setCustomDiskVerificationFailed] = useState(false);
+    const [customDiskVerificationMessage, setCustomDiskVerificationMessage] = useState(null);
+    const [dialogError, setDialogError] = useState(null);
+    const [dialogErrorDetail, setDialogErrorDetail] = useState(null);
+    const [dialogLoading, setDialogLoading] = useState(true);
+    const [diskParams, setDiskParams] = useState({});
+    const [mode, setMode] = useState(isMediaInsertion ? CUSTOM_PATH : CREATE_NEW);
+    const [validationFailed, setValidationFailed] = useState({});
+    const [verificationInProgress, setVerificationInProgress] = useState(false);
+
+    const Dialogs = useDialogs();
+    const currentMode = useRef();
+
+    const storagePools = useMemo(() => getVmStoragePools(vm), [vm]);
+    const defaultPool = storagePools.sort(sortFunction)[0];
+
+    // default true for a down VM; for a running domain, the disk is attached tentatively only
+    // must be kept false for a down VM; the value is not being changed by user
+    const initialDiskParams = useMemo(() => ({
+        busType: "virtio",
+        cacheMode: 'default',
+        connectionName: vm.connectionName,
+        device: "disk",
+        existingVolumeName: defaultPool && getDefaultVolumeName(defaultPool, vm),
+        file: "",
+        format: "",
+        permanent: !domainIsRunning(vm.state),
+        serial: "",
+        size: 1,
+        storagePool: defaultPool,
+        type: mode === CUSTOM_PATH ? "file" : "volume",
+        unit: units.GiB.name,
+        vmId: vm.id,
+        vmName: vm.name,
+        volumeName: "",
+    }), [defaultPool, mode, vm]);
+    const storagePoolName = diskParams.storagePool?.name;
 
-    componentDidMount() {
+    useEffect(() => {
         // Refresh storage volume list before displaying the dialog.
         // There are recently no Libvirt events for storage volumes and polling is ugly.
         // https://bugzilla.redhat.com/show_bug.cgi?id=1578836
-        storagePoolGetAll({ connectionName: this.props.vm.connectionName })
-                .then(() => this.setState({ dialogLoading: false, ...this.initialState }))
-                .catch(exc => this.dialogErrorSet(_("Storage pools could not be fetched"), exc.message));
-    }
-
-    validateParams() {
-        const validationFailed = {};
-
-        if (this.state.mode !== CUSTOM_PATH && !this.state.storagePoolName)
-            validationFailed.storagePool = _("Please choose a storage pool");
+        storagePoolGetAll({ connectionName: vm.connectionName })
+                .always(() => setDialogLoading(false))
+                .catch(exc => dialogErrorSet(_("Storage pools could not be fetched"), exc.message));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231113-155652-97d90b8c-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)